jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PushContainerConfigurationStep.java

```diff
@@ -16,90 +16,58 @@
-
-  private final ListenableFuture<AsyncStep<PushBlobStep>> listenableFuture;
+  private final Authorization pushAuthorization;
+  private final Image builtImage;
 
   PushContainerConfigurationStep(
-      ListeningExecutorService listeningExecutorService,
       BuildConfiguration buildConfiguration,
       ProgressEventDispatcher.Factory progressEventDispatcherFactory,
-      AuthenticatePushStep authenticatePushStep,
-      BuildImageStep buildImageStep) {
-    this.listeningExecutorService = listeningExecutorService;
+      Authorization authenticatePushStep,
+      Image builtImage) {
     this.buildConfiguration = buildConfiguration;
     this.progressEventDispatcherFactory = progressEventDispatcherFactory;
-    this.authenticatePushStep = authenticatePushStep;
-    this.buildImageStep = buildImageStep;
-
-    listenableFuture =
-        AsyncDependencies.using(listeningExecutorService)
-            .addStep(buildImageStep)
-            .whenAllSucceed(this);
+    this.pushAuthorization = authenticatePushStep;
+    this.builtImage = builtImage;
   }
 
   @Override
-  public ListenableFuture<AsyncStep<PushBlobStep>> getFuture() {
-    return listenableFuture;
-  }
-
-  @Override
-  public AsyncStep<PushBlobStep> call() throws ExecutionException {
-    ListenableFuture<PushBlobStep> pushBlobStepFuture =
-        AsyncDependencies.using(listeningExecutorService)
-            .addStep(authenticatePushStep)
-            .addStep(NonBlockingSteps.get(buildImageStep))
-            .whenAllSucceed(this::afterBuildConfigurationFutureFuture);
-    return () -> pushBlobStepFuture;
-  }
-
-  private PushBlobStep afterBuildConfigurationFutureFuture()
-      throws ExecutionException, IOException {
+  public BlobDescriptor call() throws IOException, RegistryException {
     try (ProgressEventDispatcher progressEventDispatcher =
             progressEventDispatcherFactory.create("pushing container configuration", 1);
         TimerEventDispatcher ignored =
             new TimerEventDispatcher(buildConfiguration.getEventHandlers(), DESCRIPTION)) {
-      Image image = NonBlockingSteps.get(NonBlockingSteps.get(buildImageStep));
       JsonTemplate containerConfiguration =
-          new ImageToJsonTranslator(image).getContainerConfiguration();
-      BlobDescriptor blobDescriptor = Digests.computeDigest(containerConfiguration);
+          new ImageToJsonTranslator(builtImage).getContainerConfiguration();
 
       return new PushBlobStep(
-          listeningExecutorService,
-          buildConfiguration,
-          progressEventDispatcher.newChildProducer(),
-          authenticatePushStep,
-          blobDescriptor,
-          Blobs.from(containerConfiguration));
+              buildConfiguration,
+              progressEventDispatcher.newChildProducer(),
+              pushAuthorization,
+              Digests.computeDigest(containerConfiguration),
+              Blobs.from(containerConfiguration))
+          .call();
```

Comment:
Note to reviewers: we can just directly call the core to upload a blob.

Reply:
If PushBlob isn't being thrown to the executor, is it still a step?